test/integration/query/selects.js

```diff
@@ -150,6 +150,78 @@ module.exports = function (knex) {
         });
     });
 
+    it('#4199 - adheres to optimizer hints', async function () {
```

Comment:
I don't know how to test any other hint besides `max_execution_time` in MySQL as they don't alter the result just the alhorithm.

Also I don't know how to test any of the hints used in Oracle.

Reply:
Actually, PostgreSQL just doesn't support hints and don't intend to. Could you make it throw an error when you try to use it on PG? Also I think SQLite doesn't support it either.

Reply:
Is it good to throw errors if hints are used for dialects that don't support hints? There are proxies or middlewares that may listen to the hints. Like in #1982 they needed comments for passing hints for their router. Also there are DB extensions like EDB Postgres Advanced Server which support hints while vanilla Postgres does not.

Reply:
Good point, then we shouldn't throw an error.